src/js/components/FileInput/FileInput.js

```diff
@@ -0,0 +1,257 @@
+              <Label {...theme.fileInput.label}>
+                {files.length} {messages.files || 'files'}
+              </Label>
+              <Button
+                ref={removeRef}
+                a11yTitle={messages.removeAll || 'remove all'}
+                icon={<RemoveIcon />}
+                hoverIndicator
+                onClick={event => {
+                  event.stopPropagation();
+                  setFiles([]);
+                  inputRef.current.focus();
+                }}
+              />
+            </Box>
+          )}
+          {files.length > 0 &&
+            files.length <= aggregateThreshold &&
+            files.map((file, index) => (
+              <Box
+                key={file.name}
+                direction="row"
+                align="center"
+                justify="between"
+              >
+                {renderFile ? (
+                  renderFile(file)
+                ) : (
+                  <Label
+                    weight={
+                      theme.global.input.weight ||
+                      theme.global.input.font.weight
+                    }
+                    {...theme.fileInput.label}
+                  >
+                    {file.name}
+                  </Label>
+                )}
+                <Button
+                  ref={index ? undefined : removeRef}
+                  a11yTitle={`${messages.remove || 'remove'} ${file.name}`}
+                  icon={<RemoveIcon />}
+                  hoverIndicator
+                  onClick={event => {
+                    event.stopPropagation();
+                    const nextFiles = [...files];
+                    nextFiles.splice(index, 1);
+                    setFiles(nextFiles);
+                    if (nextFiles.length === 0) inputRef.current.value = '';
+                    inputRef.current.focus();
+                  }}
+                />
+              </Box>
+            ))}
+          {!files.length && (
+            <Message {...theme.fileInput.message}>
+              {multiple
+                ? messages.dropPromptMultiple || 'Drop files here or'
+                : messages.dropPrompt || 'Drop file here or'}{' '}
+              <Anchor label={messages.browse || 'browse'} />
+            </Message>
+          )}
+          <StyledFileInput
+            ref={inputRef}
+            type="file"
+            id={id}
+            name={name}
+            multiple={multiple}
+            disabled={disabled}
+            plain
+            offset={
+              removeRef.current &&
+              removeRef.current.getBoundingClientRect().width
+            }
+            {...rest}
+            onDragOver={() => setDragOver(true)}
+            onDragLeave={() => setDragOver(false)}
+            onChange={event => {
+              event.persist();
+              const fileList = event.target.files;
+              const nextFiles = multiple ? [...files] : [];
```

Comment:
In the multiple case, if I drag some files in then decide I want to drag some more after and upload a file twice, it appears twice in the list. Is this standard behavior, or would we not want to allow duplicates?

Reply:
Good check. Updated to prevent duplicates.